File: backend/app/models/legal.py
from pydantic import BaseModel


class Citation(BaseModel):
    case_name: str
    citation: str | None
    court: str | None
    year: int | None
    url: str | None


class LegalSource(BaseModel):
    citation: Citation
    relevant_quote: str
    relevance_explanation: str

File: frontend/types/legal.ts
export interface Citation
{
	case_name: string;
	citation: string | null;
	court: string | null;
	year: number | null;
	url: string | null;
}

export interface LegalSource
{
	citation: Citation;
	relevant_quote: string;
	relevance_explanation: string;
}
